### DISCUSSION TITLE
Hi, what are the advantages of presentation in imodeljs?

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#272] [General] [answered]
Hi, I can simply use presentation by modifying ruleset, but I still can't understand the great benefits provided by this library. Can you introduce it to me? thank you

**Comment by grigasp** [comment, marked as answer]
Hi,

the general idea of the presentation APIs is to help get data from iModels in a format that's suitable to present to end-users. We split that data into 2 categories: hierarchies and content.

With hierarchy rules and hierarchy related APIs (tree data provider, etc.) it's easy to create different hierarchies without doing too much coding. The library takes care of creating and associating different kinds of nodes to specific kinds of parents.

With content rules and content related APIs you can create data that can be presented in components like table, property grid, list view, etc. The library takes care not only of formatting the values, but also loading additional related properties as required by BIS rules (aspects, element types, etc.). 

In addition, the library provides what we call "unified selection". All presentation-driven components display data of some BIS instances. Unified selection means that selecting an instance in one component also selects it in all other components.

There are also a few samples / blogs that you might find useful:
- Hierarchies: 
  - https://www.itwinjs.org/sample-showcase/?group=UI+Trees&sample=presetation-tree-sample&imodel=Metrostation+Sample
- Content:
  - https://www.itwinjs.org/sample-showcase/?group=Viewer+Features&sample=property-formatting-sample&imodel=Retail+Building+Sample
- Unified selection:
  - https://www.itwinjs.org/sample-showcase/?group=UI+Trees&sample=unified-selection-tree-sample&imodel=Metrostation+Sample
  - https://medium.com/itwinjs/demystifying-unified-selection-and-selection-scopes-88b9418c57f3
  - https://medium.com/itwinjs/hooking-3rd-party-component-into-unified-selection-c4daec69789d

Hope this answers your question. If not, please let me know what's exactly not clear and I can explain in more detail.

**Reply by shenshengyi** [reply]
Thank you very much for your answers, it helped me a lot.
